Sources/StructuralCore/Forwardable.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// A type that can be converted to and from its structural representation.
+public protocol Forwardable {
```

Comment:
Isn't this just what the `Structural` protocol provides?

Reply:
Yeah, I thought about conforming all the `StructuralCons` types to `Structural` themselves, but was concerned that was a little too meta... but if you're intrigued by that direction too, then perhaps it's worth while to explore what that might look like. :-D